File: QuizUp/QuizUp-backend/src/main/java/ma/ac/emi/projectquiz/model/Choice.java
package ma.ac.emi.projectquiz.model;

public class Choice {
    private double score;
    private String text;

    public Choice(double score, String text) {
        this.score = score;
        this.text = text;
    }

    public double getScore() {
        return score;
    }
    public String getText() {
        return text;
    }

    public void setScore(double score) {
        this.score = score;
    }

    public void setText(String text) {
        this.text = text;
    }
    
}

File: QuizUp/QuizUp-frontend/src/socketsModule/quizEntities/Choice.ts


export default class Choice {
    private score:number;
    private text:string;
    
    constructor(score:number, text:string) {
        this.score = score;
        this.text = text;
    }
    public verifyAnswer(answerText:string) {
        return answerText == this.getText();
    }

    public getScore():number {
        return this.score;
    }
    public getText = () => {
        return this.text;
    }

}